### PR TITLE
Moved codebase to TypeScript (server scripts)

### PR DESCRIPTION
**Pull request by miherlosev**
Migration guide - https://github.com/DevExpress/testcafe-hammerhead/issues/1873.

### CODE REVIEW

**Comment by miherlosev** [comment]
@testcafe-build-bot retest

File: src/processing/dom/index.ts [hunk 1]
```diff
@@ -34,40 +34,43 @@ const ELEMENT_PROCESSED = 'hammerhead|element-processed';
 const AUTOCOMPLETE_ATTRIBUTE_ABSENCE_MARKER = 'hammerhead|autocomplete-attribute-absence-marker';
 
 export default class DomProcessor {
-    constructor (adapter) {
+    adapter: any;
```
[the remark this answers is not in the document]
Comment (miherlosev) [reply src/processing/dom/index.ts]: I will apply more strict typing after moving client scripts to TypeScript.

**Comment by miherlosev** [comment]
@testcafe-build-bot retest

**Comment by miherlosev** [comment]
@testcafe-build-bot retest

**Comment by miherlosev** [comment]
FPR

**Comment by miherlosev** [comment]
> So far so good. I suggest to avoid any type where interface is known. E.g. we can use the HTMLElement type.

It will be done later.
See point 'add strict type definitions for all entities (server and client)' of migration guide (https://github.com/DevExpress/testcafe-hammerhead/issues/1873#issue-392568048)

**Comment by miherlosev** [comment]
ping @aleks-pro

File: src/processing/dom/base-dom-adapter.ts [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+        'onkeypress', 'onkeyup', 'onload', 'onmousedown', 'onmouseenter', 'onmouseleave',
+        'onmousemove', 'onmouseout', 'onmouseover', 'onmouseup', 'onmousewheel', 'onpaste', 'onreset',
+        'onresize', 'onscroll', 'onselect', 'onsubmit', 'ontextinput', 'onunload', 'onwheel',
+        'onpointerdown', 'onpointerup', 'onpointercancel', 'onpointermove', 'onpointerover', 'onpointerout',
+        'onpointerenter', 'onpointerleave', 'ongotpointercapture', 'onlostpointercapture',
+        'onmspointerdown', 'onmspointerup', 'onmspointercancel', 'onmspointermove', 'onmspointerover',
+        'onmspointerout', 'onmspointerenter', 'onmspointerleave', 'onmsgotpointercapture', 'onmslostpointercapture'
+    ];
+
+    abstract removeAttr (el: any, attr: string): void;
```
Comment (aleks-pro) [inline src/processing/dom/base-dom-adapter.ts]: FYI: there is a built-in HTMLElement type in typescript

Comment (miherlosev) [reply src/processing/dom/base-dom-adapter.ts]: https://github.com/DevExpress/testcafe-hammerhead/pull/1876#issuecomment-453951423

File: src/processing/encoding/charset.ts [hunk 3]
```diff
@@ -67,32 +59,32 @@ export default class Charset {
...
-        return this.priority === PRIORITY_LIST.BOM;
+        return this.priority === CharsetPriority.BOM;
     }
 
-    fromBOM (resBuf) {
+    fromBOM (resBuf: Buffer) {
         for (let i = 0; i < CHARSET_BOM_LIST.length; i++) {
-            if (Charset._bufferStartsWithBOM(resBuf, CHARSET_BOM_LIST[i].bom))
-                return this.set(CHARSET_BOM_LIST[i].charset, PRIORITY_LIST.BOM);
+            if (resBuf.includes(CHARSET_BOM_LIST[i].bom, 0))
```
Comment (LavrovArtem) [inline src/processing/encoding/charset.ts]: I think we need to leave the previous variant.

**Comment by miherlosev** [comment]
@testcafe-build-bot retest

**Comment by miherlosev** [comment]
@testcafe-build-bot retest

**Comment by miherlosev** [comment]
@testcafe-build-bot retest

**Comment by miherlosev** [comment]
@testcafe-build-bot retest

**Comment by miherlosev** [comment]
@testcafe-build-bot retest

**Comment by miherlosev** [comment]
@testcafe-build-bot retest

**Comment by miherlosev** [comment]
@testcafe-build-bot retest

**Comment by miherlosev** [comment]
@testcafe-build-bot retest

**Comment by miherlosev** [comment]
@testcafe-build-bot retest

**Pull request merged**